src/Shared/FileUtilities.cs

```diff
@@ -893,33 +799,29 @@ internal static bool FileExistsNoThrow(string fullPath)
         /// </summary>
         internal static bool FileOrDirectoryExistsNoThrow(string fullPath)
         {
-            fullPath = AttemptToShortenPath(fullPath);
-            if (NativeMethodsShared.IsWindows)
-            {
 #if !CLR2COMPATIBILITY
-                if (Traits.Instance.CacheFileExistence)
-                {
-                    // Possible future improvement: make sure file existence caching happens only at evaluation time, and maybe only within a build session. https://github.com/Microsoft/msbuild/issues/2306
-                    return FileExistenceCache.GetOrAdd(fullPath, NativeMethodsShared.FileExists);
-                }
-                else
-                {
-#endif
-                    return NativeMethodsShared.FileExists(fullPath);
-#if !CLR2COMPATIBILITY
-                }
+            if (NativeMethodsShared.IsWindows && Traits.Instance.CacheFileExistence)
+            {
+                fullPath = AttemptToShortenPath(fullPath);
```

Comment:
Is this necessary if `GetFileAttributes()` always does it too?

Reply:
I assumed ```FileExistenceCache``` was also being called elsewhere but looks like this is the only entry point. ```AttemptToShortenPath()``` also sneaks in ```FixFilePath()``` so it could effect the ```GetOrAdd()``` key if one caller fixes the file path and another doesn't, but so far I don't see that problem.